model/model.go

```diff
@@ -23,12 +23,13 @@ type EmbeddingModel interface {
 }
 
 type Option struct {
-	Temperature      float32
-	StreamingChannel chan ChatMessage // non chat model can also use this
-	Functions        []FunctionDefinition
-	MaxToken         int
-	IsStreaming      bool
-	Verbose          bool
+	Temperature              float32
+	StreamingChannel         chan ChatMessage // non chat model can also use this
+	Functions                []FunctionDefinition
+	AdditionalMetadataFields []string
```

Comment:
as this is not related to model, this is not the right place.
However we can keep this for now until we refactor chain to have chainOption.